pyrometheus/__init__.py

```diff
@@ -117,26 +153,70 @@ def _(poly: ct.NasaPoly2, arg_name):
 
     def gen(c, t):
         assert len(c) == 7
-        return (c[0]*log(t)
-                + c[1]*t + c[2]/2*t**2 + c[3]/3*t**3 + c[4]/4*t**4
-                + c[6])
+        return (
+            c[0] * log(t)
+            + c[1] * t
+            + c[2] / 2 * t ** 2
+            + c[3] / 3 * t ** 3
+            + c[4] / 4 * t ** 4
+            + c[6]
+        )
 
     return nasa7_conditional(p.Variable(arg_name), poly, gen)
 
-# }}}
 
+# }}}
 
 # {{{ equilibrium constants
 
-def equilibrium_constants_expr(react: ct.Reaction):
-    return 0*p.Variable("T")
+
+def equilibrium_constants_expr(sol: ct.Solution, react: ct.Reaction):
+
+    reactants = react.reactants
+    products = react.products
+
+    indices_r = [sol.species_index(sp) for sp in reactants]
+    indices_p = [sol.species_index(sp) for sp in products]
+
+    coeffs_r = np.array([reactants[sp] for sp in reactants])
+    coeffs_p = np.array([products[sp] for sp in products])
+
+    n_r = coeffs_r.sum()
+    n_p = coeffs_p.sum()
+    dn = n_p - n_r
+
+    def gen(i_r, i_p, c_r, c_p, dn, arg_name):
+        num_r = len(c_r)
+        num_p = len(c_p)
+        sum_r = sum([c_r[j] * arg_name[i_r[j]] for j in range(num_r)])
+        sum_p = sum([c_p[j] * arg_name[i_p[j]] for j in range(num_p)])
```

Comment:
```suggestion
        sum_r = sum(c_r[j] * arg_name[i_r[j]] for j in range(num_r))
        sum_p = sum(c_p[j] * arg_name[i_p[j]] for j in range(num_p))
```
[List comprehension](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) vs [generator expression](https://docs.python.org/3/reference/expressions.html?highlight=generator%20expression#generator-expressions)

Reply:
Thanks, I'll have this in mind going forward.